### PR TITLE
Refactor tests from callbacks to async await

### PR DESCRIPTION
**Pull request by AbdelrahmanHafez**

### CODE REVIEW

File: test/aggregate.test.js [hunk 1]
```diff
@@ -1277,25 +1264,23 @@ describe('aggregate: ', function() {
       const mySchema = new Schema({ name: String, qty: Number });
       mySchema.index({ qty: -1, name: -1 });
       const M = db.model('Test', mySchema);
-      M.on('index', function(error) {
+      M.on('index', async function(error) {
```
Comment (vkarpov15) [inline test/aggregate.test.js:1267]: We should replace `M.on('index', async function(error) { ... })` with `await M.init(); ...`

**Review by vkarpov15 — APPROVED** [review]
1 quick comment, otherwise LGTM :+1:

File: test/aggregate.test.js (hunk 1 above)
Comment (AbdelrahmanHafez) [reply test/aggregate.test.js:1267]: Ah yes, I've avoided modifying those tests.
`await M.init();` seems like the way to go. 👍

File: test/model.indexes.test.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ describe('model', function() {
       const IndexedModel = db.model('Test', Indexed);
       let assertions = 0;
 
-      await new Promise((resolve) => IndexedModel.on('index', () => resolve()));
+      await Indexed.init();
```
Comment (vkarpov15) [inline test/model.indexes.test.js:50]: Should be `IndexedModel.init()`

File: test/model.indexes.test.js [hunk 3]
```diff
@@ -83,33 +83,28 @@ describe('model', function() {
...
-              assertions++;
-            }
-            if (index[0] === 'blogposts._id') {
-              assertions++;
-            }
-            if (index[0] === 'blogposts.title') {
-              assertions++;
-            }
-          }
+      const mongoIndexes = await UserModel.collection.getIndexes();
```
Comment (vkarpov15) [inline test/model.indexes.test.js:88]: Looks like `getIndexes()` resolves to an object now, `Object.values(await UserModel.collection.getIndexes())` should be what you're looking for.

**Review by vkarpov15 — CHANGES_REQUESTED** [review]

File: test/model.indexes.test.js (hunk 3 above)
Comment (AbdelrahmanHafez) [reply test/model.indexes.test.js:88]: Thanks, I'll take a look at those failing tests sometime during the weekend, then we can merge and I can work on another batch on another PR.

**Review by vkarpov15 — APPROVED** [review]
Nice :+1:

**Pull request merged**